crates/pathfinder/src/state/sync.rs

```diff
@@ -97,8 +85,72 @@ pub fn sync(
                 };
                 println!("L1 reorg occurred, new L1 head is {:?}", new_head);
             }
-            SyncEvent::L2Reorg(reorg_tail) => {
+            Ok(l1::Event::QueryUpdate(block, tx)) => {
+                let update =
+                    tokio::task::block_in_place(|| L1StateTable::get(&db_conn, block.into()))
```

Comment:
`tokio::task::block_in_place()` is wrapping all the blocking code now - I'm wondering why not `spawn_blocking`? I'm thinking because it's more convenient and one does not have to manage a join handle?

Reply:
The database connection lifetime becomes difficult to specify -- since `spawn_blocking` can live forever, it tries to force the connection to be `'static` but we need the connection in other places as well. An alternative is creating a new connection each time, but that also seemed.. weird. So I just went for `block_in_place` -- not sure if there isn't some hidden penalty but I don't think so since this is technically linear code.